backend/src/services/groups-to-permissions/groups-to-permissions.service.ts

```diff
@@ -26,6 +26,58 @@ class GroupsToPermissions {
       permissionId: model.permissionId,
     };
   }
+
+  async getPermissionsByGroupId(
+    groupId: number,
+  ): Promise<GroupsToPermissionsResponseDto[]> {
+    const permissions = await this.#groupsToPermissionsRepository.getByGroupId(
+      groupId,
+    );
+
+    return permissions;
+  }
+
+  async updatePermissionsByGroupId(groupsToPermissions: {
+    groupId: number;
+    permissionIds: number[];
+  }): Promise<void> {
+    const { groupId, permissionIds } = groupsToPermissions;
+    const groupPrermissions = await this.getPermissionsByGroupId(groupId);
+    const dbPermissionIds = groupPrermissions.map(
+      (permission) => permission.permissionId,
+    );
+
+    const permissionsToCreate = permissionIds.filter((id: number) => {
+      return !dbPermissionIds.includes(id);
+    });
+    const permissionsToDelete = groupPrermissions.filter(
+      (permission: GroupsToPermissionsResponseDto) => {
+        return !permissionIds.includes(permission.permissionId);
+      },
+    );
```

Comment:
You've already mapped `dbPermissionsIds` so can use this array
```suggestion
    const permissionsToDelete = dbPermissionIds.filter((id: number) => {
      return !permissionIds.includes(id);
    });
```

Reply:
i did with `groupPermissions` so that i can delete rows by their id, else `permissionsToDelete` would contain `permissionId`s then i would also need to pass `groupId` to delete the row

Reply:
will do same as for users to groups with unique constraint